### PR TITLE
endpoints: add error handler for RequestRedirect (PROJQUAY-6724)

### PR DESCRIPTION
**Pull request by Marcusk19**
Adds an error handler for RequestRedirect exceptions raised by werkzeug to reduce noise in sentry

### CODE REVIEW

**Review by arborite-rh — COMMENTED** [review]
this looks good. should we add unit tests?

**Comment by Marcusk19** [comment]
Yeah, still working on the unit tests. It's proving trickier than I thought to mock/raise this exception properly in my test.

**Comment by Marcusk19** [comment]
@arborite-rh FYI, added a unit test for this

File: endpoints/test/test_request_redirect.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        @self.app.application.route("/raise-request-redirect")
+        def raise_request_redirect():
+            raise RequestRedirect("somepath")
+
+        # add error handler to test client
+        @self.app.application.errorhandler(RequestRedirect)
+        def handle_request_redirect(e):
+            return endpoints.decorated.handle_bad_redirect(e)
+
+        path = "/raise-request-redirect"
```
Comment (arborite-rh) [inline endpoints/test/test_request_redirect.py:35]: nit: we could create a var anywhere this path is used (here and line 26)

**Review by arborite-rh — APPROVED** [review]
lgtm! 1 nitpick comment. 6 lines of new code with 40 lines to test, thats how it usually goes 😄

**Pull request merged**